orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v1schema/TemplateMerge.java

```diff
@@ -31,16 +31,21 @@ public static PipelineTemplate merge(List<PipelineTemplate> templates) {
     for (PipelineTemplate template : templates) {
       appendSource(result, template);
 
-      // TODO rz - should validate var types are the same here or elsewhere?
+      // If any of the templates are set as protected, configurations won't be allowed to alter stages.
+      if (template.getProtect()) {
+        result.setProtect(true);
+      }
+
       result.setVariables(mergeNamedContent(result.getVariables(), template.getVariables()));
 
       mergeConfiguration(result, template);
 
-      // TODO rz - might make sense that any child template must use inject syntax / modules.
       result.setStages(mergeIdentifiable(result.getStages(), template.getStages()));
-
       result.setModules(mergeIdentifiable(result.getModules(), template.getModules()));
     }
+
+    result.setMetadata(templates.get(templates.size() - 1).getMetadata());
```

Comment:
Will add a comment to the code, but we're only interested in applying the template metadata from the last template. Maybe this won't be necessary for the sake of building up pipeline executions, though.